docker/registry/internal/ecr.go

```diff
@@ -4,14 +4,67 @@
 package internal
 
 import (
+	"context"
+	"fmt"
 	"net/http"
 	"strings"
+	"time"
 
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/smithy-go/logging"
 	"github.com/juju/errors"
 
 	"github.com/juju/juju/docker"
+	"github.com/juju/juju/tools"
 )
 
+// TODO: implement token refresh feature and integrate it into the caasmodelconfigmanager worker!!
+// Because the token used for image pulling has to be refreshed every 12hrs.
+// https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html
+
+// GetECRClient is exported for test to patch.
+var GetECRClient = getECRClient
```

Comment:
Can we pass this in to where it's used as an interface and use gomock for tests so that we don't need to patch?
We already use gomock for ECRInterface